modules/project_cleanup/main.tf

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google-beta" {
+  version = "~> 2.1"
+  project = "${var.project_id}"
+  region  = "${var.region}"
+}
+
+resource "google_project_service" "cloudfunctions" {
```

Comment:
Is there any guarantee these services will be activated before the root module is invoked?

Reply:
@aaron-lane No, there's no such guarantee. I've instead extracted this resource (and the other `google_project_service`s) to the responsibility of the calling module, and notated them as required in the `README`.